packages/smithy-aws-core/tests/unit/test_utils.py

```diff
@@ -0,0 +1,78 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  SPDX-License-Identifier: Apache-2.0
+
+import pytest
+from smithy_aws_core.utils import parse_document_discriminator, parse_error_code
+from smithy_core.documents import Document
+from smithy_core.shapes import ShapeID
+
+
+@pytest.mark.parametrize(
+    "document, expected",
+    [
+        ({"__type": "FooError"}, "com.test#FooError"),
+        ({"__type": "com.test#FooError"}, "com.test#FooError"),
+        (
+            {
+                "__type": "FooError:http://internal.amazon.com/coral/com.amazon.coral.validate/"
```

Comment:
Is there a better error shape for this? I'm a little surprised we're using the internal coral reference in our tests. I see that's also done in the protocol tests and listed on the smithy website.

Reply:
Talked to Kevin, I guess this is just a public detail with Smithy.